src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/DefaultHtmlFactsService.cs

```diff
@@ -46,56 +46,75 @@ public override bool TryGetElementInfo(SyntaxNode element, out SyntaxToken conta
+            switch (attribute)
             {
-                prefixLocation = minimizedAttributeBlock.NamePrefix.Span;
-                selectedAttributeName = minimizedAttributeBlock.Name.GetContent();
-                selectedAttributeNameLocation = minimizedAttributeBlock.Name.Span;
-                return true;
-            }
-            else if (attribute is MarkupAttributeBlockSyntax attributeBlock)
-            {
-                prefixLocation = attributeBlock.NamePrefix.Span;
-                selectedAttributeName = attributeBlock.Name.GetContent();
-                selectedAttributeNameLocation = attributeBlock.Name.Span;
-                return true;
-            }
-            else if (attribute is MarkupTagHelperAttributeSyntax tagHelperAttribute)
-            {
-                prefixLocation = tagHelperAttribute.NamePrefix.Span;
-                selectedAttributeName = tagHelperAttribute.Name.GetContent();
-                selectedAttributeNameLocation = tagHelperAttribute.Name.Span;
-                return true;
-            }
-            else if (attribute is MarkupMinimizedTagHelperAttributeSyntax minimizedAttribute)
-            {
-                prefixLocation = minimizedAttribute.NamePrefix.Span;
-                selectedAttributeName = minimizedAttribute.Name.GetContent();
-                selectedAttributeNameLocation = minimizedAttribute.Name.Span;
-                return true;
-            }
-            else if (attribute is MarkupTagHelperDirectiveAttributeSyntax tagHelperDirectiveAttribute)
-            {
-                prefixLocation = tagHelperDirectiveAttribute.NamePrefix.Span;
-                selectedAttributeName = tagHelperDirectiveAttribute.FullName;
-                var fullNameSpan = TextSpan.FromBounds(tagHelperDirectiveAttribute.Transition.Span.Start, tagHelperDirectiveAttribute.Name.Span.End);
-                selectedAttributeNameLocation = fullNameSpan;
-                return true;
-            }
-            else if (attribute is MarkupMinimizedTagHelperDirectiveAttributeSyntax minimizedTagHelperDirectiveAttribute)
-            {
-                prefixLocation = minimizedTagHelperDirectiveAttribute.NamePrefix.Span;
-                selectedAttributeName = minimizedTagHelperDirectiveAttribute.FullName;
-                var fullNameSpan = TextSpan.FromBounds(minimizedTagHelperDirectiveAttribute.Transition.Span.Start, minimizedTagHelperDirectiveAttribute.Name.Span.End);
-                selectedAttributeNameLocation = fullNameSpan;
-                return true;
-            }
-            else if (attribute is MarkupMiscAttributeContentSyntax)
-            {
-                prefixLocation = null;
-                selectedAttributeName = null;
-                selectedAttributeNameLocation = null;
-                return true;
+                case MarkupMinimizedAttributeBlockSyntax minimizedAttributeBlock:
+                    if (minimizedAttributeBlock.NamePrefix == null)
```

Comment:
Deleted my previous comment. I had mistaken this PR as solving a different issue.

If the `NamePrefix` is `null` the attribute can still be valid right? I'm not convinced it's worth "break"ing on the situation that it's `null` if we can still extract other information, i.e. it being `null` is equivalent to it being a 0 length TextSpan. That being said maybe I'm misunderstanding the root cause.

Reply:
That's a good point. The `HtmlFactsService` accepts a null `TextSpan? prefixLocation`. Updated `DirectiveAttributeCompletionItemProviderBase.cs` to also accept `TextSpan?` and added an appropriate test. The `DirectiveAttributeCompletionItemProvider` doesn't even use the output (https://github.com/dotnet/aspnetcore-tooling/blob/master/src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Completion/DirectiveAttributeCompletionItemProvider.cs#L58).